util/util.go

```diff
@@ -29,6 +32,24 @@ func GenerateUUID() string {
 	return uuid.New().String()
 }
 
+
+func Transport(pURL string) *http.Transport {
+	var tr http.Transport
+	if pURL != "" {
+		paURL, err := url.Parse(pURL)
+		if err != nil {
+			panic("Error parsing proxy URL")
+		}
+		tr = http.Transport{
+			Proxy:           http.ProxyURL(paURL),
+		}
+	} else {
```

Comment:
This `else` is unnecessary. `var tr http.Transport` already allocate the memory with a default `Transport`